typespec-extension/src/emitter.ts

```diff
@@ -118,6 +118,16 @@ export async function $onEmit(context: EmitContext<EmitterOptions>) {
     const outputPath = options["output-dir"] ?? context.emitterOutputDir;
     options["output-dir"] = getNormalizedAbsolutePath(outputPath, undefined);
 
+    if (!options["flavor"]) {
+      if (options["package-dir"]?.toLocaleLowerCase().startsWith("azure")) {
+        // Azure package
+        options["flavor"] = "Azure";
+      } else {
+        // default
+        options["flavor"] = "Azure";
+      }
+    }
+
```

Comment:
Coud we move this block to before
https://github.com/Azure/autorest.java/blob/a20ba150b6c4c4cc57c6de6e5dd0c8b6fdc6b687/typespec-extension/src/emitter.ts#L111-L112

This way, it make sure in code-model-builder, the `options.flavor` is already resolved.